eureka-client/src/main/java/com/netflix/discovery/shared/resolver/ResolverUtils.java

```diff
@@ -41,13 +41,40 @@
 
     private static final Logger logger = LoggerFactory.getLogger(ResolverUtils.class);
 
-    private static final String LOCAL_IPV4_ADDRESS = SystemUtil.getServerIPv4();
-
     private static final Pattern ZONE_RE = Pattern.compile("(txt\\.)?([^.]+).*");
 
+    private static Randomizer RANDOMIZER = new Randomizer() {
+
+        private final String LOCAL_IPV4_ADDRESS = SystemUtil.getServerIPv4();
```

Comment:
Currently the local address is derived from the local InstanceInfo, which makes testing easier and keeps the source of truth for such information in once place.  For consistency I think it's best to continue using the local InstanceInfo.